solidity/contracts/converter/types/liquidity-pool-v1/LiquidityPoolV1Converter.sol

```diff
@@ -69,6 +82,25 @@ contract LiquidityPoolV1Converter is LiquidityPoolConverter {
         emit Activation(converterType(), anchor, true);
     }
 
+    /**
+      * @dev returns the rate of 1 `_token1` in `_token2` units
+      * note that the rate can only be queried for the reserves
+      *
+      * @param _token1  reserve token to get the rate of one unit of
+      * @param _token2  reserve token to get the rate of one `_token1` unit in
+      *
+      * @return rate (numerator)
+      * @return rate (denominator)
+    */
+    function tokensRate(IERC20Token _token1, IERC20Token _token2) public view returns (uint256, uint256) {
+        uint256 token1Balance = reserves[_token1].balance;
+        uint256 token2Balance = reserves[_token2].balance;
+        uint32 token1Weight = reserves[_token1].weight;
+        uint32 token2Weight = reserves[_token2].weight;
+

```

Comment:
If called a lot, then perhaps it is better to do:
```
Reserve storage reserve1 = reserves[_token1];
Reserve storage reserve2 = reserves[_token2];
uint256 token1Balance = reserve1.balance;
uint256 token2Balance = reserve2.balance;
uint32 token1Weight = reserve1.weight;
uint32 token2Weight = reserve2.weight;
```

Reply:
done